### PR TITLE
[Docs] Add files for KD algo DFND

### PR DESCRIPTION
**Pull request by Cbtor**
## Motivation

DFND (Learning Student Networks in the Wild) is a famous algorithm for data-free noisy distillation. It could achieve comparable accuracy with supervised training model using only noisy wild data. The [paper](https://openaccess.thecvf.com/content/CVPR2021/papers/Chen_Learning_Student_Networks_in_the_Wild_CVPR_2021_paper.pdf) is published on CVPR 2021.

## Modification

Add files for DFND algorithm.

## Use cases (Optional)

```bash
python tools/train.py configs/distill/mmcls/dfnd/dfnd_logits_resnet34_resnet18_8xb32_cifar10.py
```

### CODE REVIEW

**Comment by Cbtor** [comment]
I need some help. There are two `mdformat` check in lint, and the code fails the first one but pass the second. What is the difference between them? Meanwhile, I already use `docformatter`(version==1.7.5) tool to format my code, but it still fail the check. Could you please help me to find out how to pass the checks? @HIT-cwh @humu789 

```
mdformat.................................................................Failed
- hook id: mdformat
- files were modified by this hook
docformatter.............................................................Failed
- hook id: docformatter
- files were modified by this hook
mdformat.................................................................Passed
```

**Comment by c122-ode** [comment]
fork
